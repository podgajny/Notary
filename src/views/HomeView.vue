<script setup lang="ts">
import { onMounted } from 'vue'
import NoteEditor from '../components/NoteEditor.vue'
import NoteList from '../components/NoteList.vue'
import { useNotesStore } from '../stores/notes.store'

// Inicjalizacja store
const store = useNotesStore()

onMounted(async () => {
  // Za≈Çaduj notatki i drafty przy starcie aplikacji
  await store.initialize()
})
</script>

<template>
  <main class="container mx-auto py-8 px-4 space-y-8">
    <!-- Edytor notatek -->
    <section>
      <NoteEditor />
    </section>

    <!-- Lista notatek -->
    <section>
      <NoteList />
    </section>
  </main>
</template>
