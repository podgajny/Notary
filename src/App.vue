<template>
  <div id="app" class="min-h-screen bg-background">
    <div class="container mx-auto py-8">
      <h1 class="text-3xl font-bold text-center mb-8">Notary - Notatki</h1>
      <HomeView />
    </div>
  </div>
</template>

<script setup lang="ts">
import { onMounted } from 'vue'
import { useNotesStore } from './stores/notes.store'
import HomeView from './views/HomeView.vue'

// Inicjalizuj store przy starcie aplikacji
const notesStore = useNotesStore()

onMounted(async () => {
  await notesStore.initialize()
})
</script>