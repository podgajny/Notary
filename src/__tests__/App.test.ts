import { describe, it, expect, beforeEach, vi } from "vitest";
import { mount } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import { createRouter, createWebHistory } from "vue-router";
import App from "../App.vue";
import HomeView from "../views/HomeView.vue";

// Mock the database functions
vi.mock("../lib/db", () => ({
  getNotes: vi.fn().mockResolvedValue([]),
  setNotes: vi.fn().mockResolvedValue(undefined),
  DbError: class MockDbError extends Error {
    code: string;
    constructor(code: string, message: string) {
      super(message);
      this.code = code;
    }
  },
}));

describe("App", () => {
  let router: any;
  let pinia: any;

  beforeEach(() => {
    // Create fresh instances for each test
    pinia = createPinia();
    setActivePinia(pinia);

    router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: "/",
          name: "home",
          component: HomeView,
        },
      ],
    });
  });

  it("powinien renderować się bez błędów", () => {
    const wrapper = mount(App, {
      global: {
        plugins: [pinia, router],
      },
    });
    expect(wrapper.exists()).toBe(true);
  });

  it("powinien mieć odpowiednią strukturę HTML", () => {
    const wrapper = mount(App, {
      global: {
        plugins: [pinia, router],
      },
    });

    // Sprawdź czy główny div ma id="app"
    expect(wrapper.attributes("id")).toBe("app");
    expect(wrapper.findComponent({ name: "RouterView" }).exists()).toBe(true);
  });

  // Test snapshot - sprawdza czy struktura się nie zmieniła
  it("powinien pasować do snapshot", () => {
    const wrapper = mount(App, {
      global: {
        plugins: [pinia, router],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
