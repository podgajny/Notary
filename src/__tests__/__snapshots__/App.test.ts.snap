// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > powinien pasować do snapshot 1`] = `
"<div id="app" class="min-h-screen bg-gray-50 py-8">
  <!-- Główny komponent wyświetlający notatkę -->
  <div data-v-6a099708="" class="max-w-4xl mx-auto p-6 bg-white rounded-lg shadow-sm">
    <!-- Tytuł notatki -->
    <h1 data-v-6a099708="" class="text-3xl font-bold text-gray-900 mb-6 leading-tight">TEST: Zmieniony tytuł notatki - wersja testowa2</h1><!-- Treść notatki -->
    <div data-v-6a099708="" class="text-base text-gray-700 leading-relaxed space-y-4">Dobry wieczór. Coś się... coś się popsuło i nie było mnie słychać, to powtórzę jeszcze raz. Wynik wyborczy KWW Stonogi to jest jakaś porażka, ja myślę że to polskie, kurwa głupie społeczeństwo, te kurwa banda imbecyli która głosowała na tych kurwów karakanów z PiSu to jest jakieś kurwa nieporozumienie. Jeśli tyle dla was znaczy... jeśli tyle dla was znaczy, ludzie takie zaangażowanie społeczne, jak moje gdzie postawiłem moją rodzinę, moje życie prywatne, biznes, wszystko inne i dla was tylko znaczyło to pierdolone 70 czy 80 tysięcy głosów, to was powinno się jebać kurwa, tak się powinno was jebać, jak tylko was może jebać PiS.</div>
  </div>
</div>"
`;
